protocols/v2/binary-sv2/no-serde-sv2/codec/src/lib.rs

```diff
@@ -19,6 +19,9 @@
 //! Seq0255  <-> SEQ0_255[T]
 //! Seq064K  <-> SEQ0_64K[T]
 //! ```
+
+#![cfg_attr(feature = "no_std", no_std)]
```

Comment:
We can now use `#![no_std]`, and I'm confident that we are not using any `std`-compliant methods in `binary_codec_sv2`  in any of the top-level crates that depend on it. Since almost all of our crates are `no_std`-compliant, makes sense to have no-serde-sv2 to no-std only. cc: @rrybarczyk

Reply:
I asked about this in this [comment](https://github.com/stratum-mining/stratum/pull/1230#issuecomment-2441362158).

Current `binary_codec_sv2` has a broken `std` support, do we want to remove it or fix it ?

Look like you propose to remove it. It is also my preference.

Reply:
I also think we should enforce `no_std` on the `protocols` crates. So let's remove.

Reply:
This PR looks good to me and is ready to go. We can enforce the `no-std` restriction during our protocol refactoring effort, once we have more concrete reasoning for removing `std`-related imports. However, I’m fine with either approach.

Reply:
Agree. Let's confirm in the dev meet before we approve.